Sources/Signal.swift

```diff
@@ -316,10 +316,15 @@ public final class Signal<Value, Error: Swift.Error> {
 					if case let .alive(snapshot) = s.state {
 						var observers = snapshot.observers
 						observers.remove(using: token)
-						s.state = .alive(AliveState(observers: observers,
-						                            retaining: observers.isEmpty ? nil : self))
+
+						withExtendedLifetime(snapshot.retaining) {
+							s.state = .alive(AliveState(observers: observers,
+							                            retaining: observers.isEmpty ? nil : self))
```

Comment:
Thoughts on extracting this to a local `newState` so we can remove the awkward formatting over two lines?